### PR TITLE
Support @matchlabels annotating rule

### PR DESCRIPTION
**Pull request by dulltz**
https://github.com/plexsystems/konstraint/issues/93

```rego
# @matchlabels app=a team=b
package foo

import data.lib.core
...
```
↓
```yaml
apiVersion: constraints.gatekeeper.sh/v1beta1
kind: Foo
metadata:
  name: foo
spec:
  match:
    labelSelector:
      matchLabels:
        app: a
        team: b
```

### CODE REVIEW

File: internal/rego/matchers.go [hunk 1]
```diff
@@ -25,13 +28,23 @@ type KindMatcher struct {
...
 	var matchers Matchers
 	for _, comment := range r.comments {
 		if strings.HasPrefix(comment, "@kinds") {
 			matchers.KindMatchers = getKindMatchers(comment)
 		}
+		if strings.HasPrefix(comment, "@matchlabels") {
+			var err error
+			matchers.MatchLabelsMatcher, err = getMatchLabelsMatcher(comment)
+			if err != nil {
+				log.Fatal(err)
```
Comment (jalseth) [inline internal/rego/matchers.go:45]: Opinion: Functions in packages should not end execution, they should only return errors back to the caller. This function should be updated to also return an `error`.

File: docs/constraint_creation.md [hunk 2]
```diff
@@ -76,6 +76,17 @@ pod_has_hostipc {
 
 The comment block is also what is used when generating documentation via the `doc` command.
 
+### Annotating rules for matchers
+
+- `@kinds` generates `constraint.spec.match.kinds`. The allowed format is whitespace-separated `<group>/<kind>` list.
```
Comment (jalseth) [inline docs/constraint_creation.md:81]: nit: I would prefer `whitespace` changed to `space` to clarify we don't support tabs.

File: docs/constraint_creation.md [hunk 3]
```diff
@@ -76,6 +76,17 @@ pod_has_hostipc {
...
+    ```
+    # @kinds apps/DaemonSet apps/Deployment apps/StatefulSet core/Pod
+    ```
+- `@matchlabels` generates `constraint.spec.match.labelSelector.matchLabels`. The allowed format is whitespace-separated `<label-key>=<label-val>` list.
```
Comment (jalseth) [inline docs/constraint_creation.md:85]: nit: I would prefer `whitespace` changed to `space` to clarify we don't support tabs.

**Review by jalseth — CHANGES_REQUESTED** [review]
Thanks for the PR! This is a great start, there are just a few things I'd like to see updated. Additionally, the `doc` command needs to be updated as well, see https://github.com/plexsystems/konstraint/blob/b39a548fec97f387254036bef25e287b3727564e/internal/commands/document.go#L132

File: internal/rego/matchers.go (hunk 1 above)
Comment (dulltz) [reply internal/rego/matchers.go]: I fixed it!

File: docs/constraint_creation.md (hunk 2 above)
Comment (dulltz) [reply docs/constraint_creation.md]: Certainly it looks better so I fixed it

File: docs/constraint_creation.md (hunk 3 above)
Comment (dulltz) [reply docs/constraint_creation.md]: Certainly it looks better so I fixed it.

**Comment by jalseth** [comment]
@dulltz Thanks for making those changes! I see you have opted to create a `**MatchLabels:** *` entry in the documentation if there are no match labels set. I would prefer if this was not included if MatchLabels item was empty. Does that sound OK to you?

**Comment by dulltz** [comment]
@jalseth 
Sounds good! I'll try it by text/template `{{ if }}` action.

**Comment by jalseth** [comment]
LGTM! Thanks again for the contribution!

**Pull request merged**